Feature/tickets and activities locale option

ray-001 opened the pull request:
- Added locale option to queries
-  Changed PartnerBookingResponse and corresponding classes to PartnerRelay
- Separated updatedBookings call as this is not using a relay
- Added company result in partner
- Added company call
- Added activity call
- Added ticket call
- Added relay function for tickets from partner
- Added call for bulk operation for ticket timeslots
- Added complete pending booking call

ray-001 committed:
- Changed PartnerBookingResponse and corresponding classes to PartnerRelay
- Separated updatedBookings call as this is not using a relay
- Added company result in partner
- Added company call
- Added activity call
- Added ticket call
- Added relay function for tickets from partner
- Added call for bulk operation for ticket timeslots
- Added complete pending booking call
- Updated docs to include new calls

ray-001 committed:
added locale option to querybuilder for fields that have multiple options

ray-001 committed:
added accessibility for constants

ray-001 committed:
Merge branch 'feature/tickets-and-activities' into feature/tickets-and-activities-locale-option

# Conflicts:
#	src/ApiClient.php

ray-001 committed:
constants on top

ray-001 committed:
Merge branch 'feature/tickets-and-activities' into feature/tickets-and-activities-locale-option

# Conflicts:
#	src/ApiClient.php

ray-001 committed:
Added update.md

ray-001 committed:
small bug fix for complete pending booking

KellydeHaan commented on src/Model/TimeslotInput.php:
```diff
@@ -0,0 +1,193 @@
+
+use DateTimeInterface;
+
+class TimeslotInput implements InputInterface
```
Ik begrijp dat het voor een gebruiker makkelijk kan zijn om met een leeg object te kunnen beginnen, en die dan stapgewijs te kunnen vullen, maar ik zie in deze class niet heel duidelijk terug welke velden verplicht zijn, en welke niet.

ray-001 replied:
Ik zal de nullable waardes uit de constructor en de property typehintings halen. Verder zal ik ook even aangeven in de docblock dat het required is. In principe geeft graphQL een error als je dit verkeerd doet, dus een validator zou denk ik niet nodig hoeven te zijn

KellydeHaan commented on src/Model/Translation.php:
```diff
@@ -0,0 +1,71 @@
+    /**
+     * @var string|null
+     */
+    private $label;
```
Ik betwijfel of `$label` thuishoort in deze superclass: `$locale` is de gemene deler, maar het vertaalde veld heet niet altijd `$label`

ray-001 replied:
Ik zal de class renamen dan naar translationlabel, verkeerde aanname gedaan nadat ik het alleen op labels zag.

KellydeHaan commented on src/Model/TripPricing.php:
```diff
@@ -59,7 +59,7 @@ public function __construct(
     }
```
Ook hier is volgens mij niet per se duidelijk welke velden niet nullable zijn aan de Travelbase kant (in dit geval geen). We zouden kunnen overwegen de Symfony validator toe te voegen (al maakt dat de library natuurlijk wel weer een stukje groter dan strict noodzakelijk)

ray-001 replied:
Ja, deze laat ik voor wat het is. Dit is als het goed is al in gebruik en wil het wel backwards compatible houden :-) zal het nog wel in de docblock erbij zetten.

ray-001 committed:
PR feedback

Commit:
Fix
- Timeslot start and end need to have date and time

Change
- No constant locales (since Travelbase can have any locale)
- Default locale set to Dutch

Commit:
Change:
- added ticket number

redboer commented:
@ray-001 Ik heb een commit gedaan, omdat er nog iets mis ging bij de timeslots. Kan je mijn commit nog even nagaan? Ook lijkt het een goed idee als je een composer update doet. Hij geeft naar nu een melding over.

ray-001 committed:
cleanup unused imports, updated lockfile, added language code iso reguirements in readme
